apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx

```diff
@@ -0,0 +1,171 @@
+import { useEffect } from 'react'
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+  useZodForm
+} from '@harnessio/canary'
+import { z } from 'zod'
+import { useListBranchesQuery } from '@harnessio/code-service-client'
+import { CreateFormType } from '../../../types'
+
+interface PipelineCreateFormProps {
+  onCancel: () => void
+  onSubmit: (formValues: CreateFormType) => Promise<void>
+}
+
+const createPipelineSchema = z.object({
+  name: z.string().min(1, { message: 'Pipeline name is required' }),
+  branch: z.string().min(1, { message: 'Branch name is required' }),
+  yamlPath: z.string().min(1, { message: 'YAML path is required' }),
+  storeInGit: z.boolean().optional(),
+  cloneRepo: z.boolean().optional()
+})
+
+export function PipelineCreateForm({ onCancel, onSubmit }: PipelineCreateFormProps) {
+  const form = useZodForm({
+    schema: createPipelineSchema,
+    defaultValues: {
+      name: '',
+      branch: 'main',
+      yamlPath: ''
+    }
+  })
+
+  const { data, isLoading } = useListBranchesQuery({ repo_ref: 'MyNewProject/MyRepo/+', queryParams: {} })
+
+  const { watch, setValue, clearErrors, trigger } = form
```

Comment:
can we destructure inline on line 41? makes code more readable imo

Reply:
It is not props, it is form from `useZodForm`. As it required to be passed to `Form` component, we cannot destructure it.